pkgs/applications/editors/emacs/build-support/generic.nix

```diff
@@ -97,4 +97,4 @@ stdenv.mkDerivation (finalAttrs: ({
   '' + postInstall;
```

Comment:
Almost all other attrs are overridden by user-defined values.  Applying that pattern to `postInstall` means changing `postInstall = '' .... '' + postInstall` to `postInstall = args.postInstall or '' ... ''`, which means if users define `postInstall`, their package never does native compilation.  I think this is very bad.  However, inconsistence is also not good.  WDYT?

Reply:
Well, well, well...

Searching some packages, I have found that e.g. `copyDesktopItems` injects itself at postInstallHooks. 

```nix
postInstallHooks+=(copyDesktopItems)
# rest of code
```

It implies that post installation hooks are "queued" but not destroyed/substituted.
(And it explains why we need to never forget the `runHook pre<phase-name>`...)

That being said, curiously the consistency is not in substituting pre/post phases, but in queueing them. 
So I vote for `+`

Reply:
Note that there are [two kinds][0] of preInstall hooks, one is defined in nix files with the attr `preInstall`, others are defined as an array called `preInstallHooks` in [setup hooks][1].  For the first kind of hook, it is indeed destroyed/substituted.  For the second kind of hook, generally it is queued using `preInstallHooks+=(...)`.

[0]: https://github.com/NixOS/nixpkgs/blob/c3aa7b8938b17aebd2deecf7be0636000d62a2b9/pkgs/stdenv/generic/setup.sh#L173
[1]: https://nixos.org/guides/nix-pills/20-basic-dependencies-and-hooks#setup-hooks

Reply:
Another question, which one of the two orders should be used for `+`, `postInstall = '' .... '' + postInstall` or `postInstall = postInstall + '' .... ''`?